miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/RealMiniApp.kt

```diff
@@ -186,6 +200,19 @@ internal class RealMiniApp(
         } else checkToDownloadManifest(appId, versionId, cachedManifest)
     }
 
+    @VisibleForTesting
+    internal fun verifyCachedManifest(appId: String, versionId: String) {
+        val manifestFile = downloadedManifestCache.getManifestFile(appId)
+        if (manifestFile != null && manifestVerifier.verify(appId, manifestFile)) {
+            val customPermissions = miniAppCustomPermissionCache.readPermissions(appId)
+            val manifestPermissions = downloadedManifestCache.getAllPermissions(customPermissions)
+            miniAppCustomPermissionCache.removePermissionsNotMatching(appId, manifestPermissions)
+
+            if (downloadedManifestCache.isRequiredPermissionDenied(customPermissions))
+                throw RequiredPermissionsNotGrantedException(appId, versionId)
+        } else throw MiniAppNotFoundException(MiniAppDownloader.MINIAPP_NOT_FOUND_OR_CORRUPTED)
+    }
```

Comment:
This is pretty much the same as the code in `verifyManifest`. Could you create a common method to reduce duplicate code?